src/scripts/core/models/grid.ts

```diff
@@ -0,0 +1,57 @@
+import { Container } from "pixi.js";
+import { Column } from "./column";
+import { eventEmitter } from "../../../event-emitter";
+
+export class Grid extends Container {
+
+    private gridSize: number;
+    public columns: Column[];
+    private activeToken: number;
+
+
+    constructor (gridSize: number, availWidth: number) {
+        super ()
+
+        eventEmitter.on('clickCheck', this.clickCheck)
+        eventEmitter.on('tokenFirstClicked', this.swapPrepare);
+        eventEmitter.on('tokenSecondClicked', this.swapCheck);
+
+        this.gridSize = gridSize;
+        this.columns = [];
+        this.activeToken = 0;
+
+        for(var i = 0; i < this.gridSize; i++) {
+            const newColumn = new Column(i, this.gridSize, availWidth, availWidth);
+            newColumn.x = (availWidth / (gridSize/i));
+            this.columns.push(newColumn);
+            this.addChild(newColumn);
+        }
+        
+        this.position.set(this.columns[0].tokens[0].width/2, this.columns[0].tokens[0].height/2);
```

Comment:
As a general rule it's faster to multiply rather than divide, so it would be better to * 0.5 rather than / 2. (not that it would make much difference in this instance!)